Reorganize the installation documents. (#6877)
dmlc/xgboost: doc/install.rst

##################
Installation Guide
##################

XGBoost provides binary packages for some language bindings.  The binary packages support
the GPU algorithm (``gpu_hist``) on machines with NVIDIA GPUs. Please note that **training
with multiple GPUs is only supported for Linux platform**. See :doc:`gpu/index`.  Also we
have both stable releases and nightly builds, see below for how to install them.  For

> building from source, checkout :doc:`here </build>`.

Reviewer: ```suggestion
building from source, visit :doc:`this page </build>`.
```

.. contents:: Contents

Stable Release
==============

Python
------

> Pre-built binary are uploaded to Pypi for each release.  Supported platforms are Linux
> with x86_64, aarch, Windows x86_64 and OSX.

Reviewer: ```suggestion
Pre-built binary are uploaded to PyPI (Python Package Index) for each release.  Supported platforms are Linux (x86_64, aarch64), Windows (x86_64) and MacOS (x86_64).
```

.. code-block:: bash

  pip install xgboost

You might need to run the command with ``--user`` flag or use ``virtualenv`` if you run
into permission errors.  Python pre-built binary capability for each platform:

.. |tick| unicode:: U+2714
.. |cross| unicode:: U+2718

+-------------------+---------+----------------------+
| Platform          | GPU     | Multi-Node-Multi-GPU |
+===================+=========+======================+
| Linux x86_64      | |tick|  |  |tick|              |
+-------------------+---------+----------------------+
| Linux aarch64     | |cross| |  |cross|             |
+-------------------+---------+----------------------+

> | OSX               | |cross| |  |cross|             |

Reviewer: ```suggestion
| MacOS             | |cross| |  |cross|             |
```

+-------------------+---------+----------------------+
| Windows           | |tick|  |  |cross|             |
+-------------------+---------+----------------------+

R
-

* From CRAN:

  .. code-block:: R

    install.packages("xgboost")

  .. note:: Using all CPU cores (threads) on Mac OSX

     If you are using Mac OSX, you should first install OpenMP library (``libomp``) by running

     .. code-block:: bash

        brew install libomp

     and then run ``install.packages("xgboost")``. Without OpenMP, XGBoost will only use a
     single CPU core, leading to suboptimal training speed.

* We also provide **experimental** pre-built binary on Linux x86_64 with GPU support.
  Download the binary package from the Releases page. The file name will be of the form
  ``xgboost_r_gpu_linux_[version].tar.gz``. Then install XGBoost by running:

  .. code-block:: bash

    # Install dependencies

>     R -q -e "install.packages(c('data.table', 'magrittr', 'jsonlite', 'remotes'))"

Reviewer: ```suggestion
    R -q -e "install.packages(c('data.table', 'magrittr', 'jsonlite'))"
```

It turns out that `remotes` is not necessary.

        <version>latest_version_num</version>
    </dependency>
    <dependency>
        <groupId>ml.dmlc</groupId>
        <artifactId>xgboost4j-spark_${scala.binary.version}</artifactId>
        <version>latest_version_num</version>
    </dependency>
  </dependencies>

.. code-block:: scala
  :caption: sbt

  libraryDependencies ++= Seq(
    "ml.dmlc" %% "xgboost4j" % "latest_version_num",
    "ml.dmlc" %% "xgboost4j-spark" % "latest_version_num"
  )

This will check out the latest stable version from the Maven Central.

For the latest release version number, please check `release page <https://github.com/dmlc/xgboost/releases>`_.

To enable the GPU algorithm (``tree_method='gpu_hist'``), use artifacts ``xgboost4j-gpu_2.12`` and ``xgboost4j-spark-gpu_2.12`` instead (note the ``gpu`` suffix).

.. note:: Windows not supported in the JVM package

  Currently, XGBoost4J-Spark does not support Windows platform, as the distributed training algorithm is inoperational for Windows. Please use Linux or MacOS.

Nightly Build
=============

Python
------

Nightly builds are available. You can go to `this page <https://s3-us-west-2.amazonaws.com/xgboost-nightly-builds/list.html>`_,

> find the wheel with the commit id you want and install it with pip:

Reviewer: ```suggestion
find the wheel with the commit ID you want and install it with pip:
```
